components/auth/org.wso2.carbon.auth.user.store/src/main/java/org/wso2/carbon/auth/user/store/internal/ConnectorDataHolder.java

```diff
@@ -70,23 +64,40 @@ public DataSource getDataSource(String dataSourceName) throws DataSourceExceptio
 
     public void setDataSourceService(DataSourceService dataSourceService) {
         this.dataSourceService = dataSourceService;
+        UserStoreConfiguration config = ServiceReferenceHolder.getInstance().getAuthConfiguration()
+                .getUserStoreConfiguration();
+        if (UserStoreConstants.JDBC_CONNECTOR_TYPE.equals(config.getConnectorType())) {
+            if (dataSourceService == null) {
+                log.warn("Admin user adding skip, since datasource is not configured.");
+                return;
+            }
+        }
 
         //adding default admin user
-        char[] password = "admin".toCharArray();
-        String user = "admin";
+        char[] password = config.getSuperUserPass().toCharArray();
+        String user = config.getSuperUser();
         UserStoreConnector connector = UserStoreConnectorFactory.getUserStoreConnector();
-        UserStoreConfiguration config = new UserStoreConfiguration();
-        Map<String, Object> properties = new HashMap<>();
-        properties.put(JDBCConnectorConstants.DATA_SOURCE, UserStoreConstants.DATASOURCE_WSO2UM_DB);
-        properties.put(JDBCConnectorConstants.DATABASE_TYPE, "MySql");
-        config.setProperties(properties);
+
         try {
             connector.init(config);
         } catch (UserStoreConnectorException e) {
             log.error("Error occurred while init UserStoreConnector", e);
+            return;
         }
         List<Attribute> attributeList = new ArrayList<>();
         attributeList.add(new Attribute(UserStoreConstants.CLAIM_USERNAME, user));
+        try {
+            String uid = connector.getConnectorUserId(UserStoreConstants.CLAIM_USERNAME, user);
+            if (uid != null) {
+                log.info("default user already exist");
+                return;
+            }
+        } catch (UserNotFoundException e) {
```

Comment:
log with e?

Reply:
should these info logs convert to debug logs?

Reply:
log and throw the exception. I don't think this user store manger use only in out rest api layer

Reply:
this is temp code to add default user to user store. I think this user need to be added using SCIM